backend/src/packages/drivers/driver.service.ts

```diff
@@ -0,0 +1,158 @@
+  }
+
+  public async findById(id: number): Promise<DriverEntityT | null> {
+    const [driver = null] = await this.driverRepository.find({ id });
+
+    return driver ? DriverEntity.initialize(driver).toObject() : null;
+  }
+
+  public async findAllByBusinessId(
+    businessId: number,
+  ): Promise<DriverGetAllResponseDto> {
+    const items = await this.driverRepository.findAllByBusinessId(businessId);
+
+    return {
+      items: items.map((it) => it.toObject()),
+    };
+  }
+
+  public async create({
+    payload,
+    id,
+  }: DriverCreatePayload): Promise<DriverAddResponseWithGroup> {
+    const { password, email, lastName, firstName, phone, driverLicenseNumber } =
+      payload;
+
+    const { result: doesDriverExist } = await this.driverRepository.checkExists(
+      {
+        driverLicenseNumber,
+      },
+    );
+
+    if (doesDriverExist) {
+      throw new HttpError({
+        status: HttpCode.BAD_REQUEST,
+        message: HttpMessage.DRIVER_ALREADY_EXISTS,
+      });
+    }
+    const group = await this.groupService.findByKey(UserGroupKey.DRIVER);
+
+    if (!group) {
+      throw new HttpError({
+        message: HttpMessage.INVALID_GROUP,
+        status: HttpCode.BAD_REQUEST,
+      });
+    }
+
+    const user = await this.userService.create({
+      password,
+      email,
+      lastName,
+      firstName,
+      phone,
+      groupId: group.id,
+    });
+
+    const driver = await this.driverRepository.create(
+      DriverEntity.initializeNew({
+        driverLicenseNumber,
+        businessId: id,
+        userId: user.id,
+      }),
+    );
+
+    const driverObject = driver.toObject();
+
+    return { ...user, group, driver: driverObject };
+  }
+
+  public async update({
+    driverId,
+    payload,
+  }: DriverUpdatePayload): Promise<DriverUpdateResponseDto> {
+    const { password, email, lastName, firstName, phone, driverLicenseNumber } =
+      payload;
+
+    const foundDriver = await this.findById(driverId);
+
+    if (!foundDriver) {
+      throw new HttpError({
+        message: HttpMessage.DRIVER_DOES_NOT_EXIST,
+        status: HttpCode.NOT_FOUND,
+      });
+    }
+
+    const user = await this.userService.update(foundDriver.userId, {
+      password,
+      email,
+      lastName,
+      firstName,
+      phone,
+    });
+
+    const driver = await this.driverRepository.update({
+      id: foundDriver.id,
+      payload: { driverLicenseNumber },
+    });
+
+    const driverObject = driver.toObject();
+
+    return { ...user, driver: driverObject };
+  }
+
+  public async delete(id: number): Promise<boolean> {
+    const foundDriver = await this.findById(id);
+
+    if (!foundDriver) {
+      throw new HttpError({
+        message: HttpMessage.DRIVER_DOES_NOT_EXIST,
+        status: HttpCode.NOT_FOUND,
+      });
+    }
+
+    const isDriverDeleted = await this.driverRepository.delete(id);
+
+    if (isDriverDeleted) {
+      return await this.userService.delete(foundDriver.userId);
+    } else {
```

Comment:
actually there can be also problems, for example
you deleted driver, and then try to delete user, but this driver has connections with trucks, so you will need to delete firstly all these connections and the delete user. 
so, I suppose soft delete will be better for these decision, but not in this pr, because it will be too large
leave a FIXME note and then we can create a separate ticket for this purpose ( actually when we will need a driver deletion flow)

Reply:
Yes, we have already discussed this, and we decided to make it a separate task